liquidctl/config/load.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import os
+import sys
+
+def get_config_files(file=None, appname='liquidctl', **kwargs):
+    """
+    This will get all the potential file paths on the system for where the config file
+    can be
+
+    This will also handle the -f overrides
+    """
+    files = []
+    if file:
+        files.append(file)
+
+    if sys.platform == 'win32':
+        files.append(os.path.join(os.getenv('APPDATA'), appname, 'config.toml'))
+    elif sys.platform == 'darwin':
+        files.append(os.path.expanduser(os.path.join('~/Library/Application Support', appname, 'config.toml')))
+        files.append(os.path.expanduser(os.path.join('~', f'.{appname}.toml')))
+    elif sys.platform == 'linux':
+        XDG_CONFIG_HOME = os.getenv('XDG_CONFIG_HOME')
+        XDG_CONFIG_DIRS = os.getenv('XDG_CONFIG_DIRS')
+
+        # treat all other platforms as *nix and conform to XDG basedir spec
+        if XDG_CONFIG_HOME:
+            files.append(os.path.join(XDG_CONFIG_HOME, appname, 'config.toml'))
+
+        files.append(os.path.expanduser(os.path.join('~', '.config', appname, 'config.toml')))
+        files.append(os.path.expanduser(os.path.join('~', f'.{appname}.toml')))
```

Comment:
These should not override a set XDG_CONFIG_HOME.  I think the latter is also use to _change_ the effective config dir for a particular program.

Reply:
ya these are both fallbacks from the `XDG_CONFIG_HOME` folder, I think I may be misunderstanding what you are saying though

Reply:
I think that, if XDG_CONFIG_HOME is set and non-empty, then `~/liquidctl/config.toml` and `~/config.toml` should *not* be checked.  XDG_CONFIG_HOME should completely override the _location_ of the user's settings, even if no config is actually found in the XDG location.

Reply:
> `~/liquidctl/config.toml` and `~/config.toml`

Neither of those are checked it is. `~/.config/liquidctl/config.toml` and `~/.liquidctl.toml`


> I think that, if `XDG_CONFIG_HOME` is set and non-empty, then .... should not be checked 

I disagree with that, `XDG_CONFIG_HOME` would be a variable set that would apply to all applications and I may have some files there for other applications but maybe not this one. I think that the fallback locations should still be checked.

Reply:
I messed up the names... I meant the correct ones.

---

I would like to make two arguments against always checking `~/.config` and ~/.liquidctl.toml`:

> There is *a single* base directory relative to which user-specific configuration files should be written. This directory is defined by the environment variable `$XDG_CONFIG_HOME`. 
> [ ... ]
> `$XDG_CONFIG_HOME` *defines the* base directory relative to which user specific configuration files should be stored. *If* `$XDG_CONFIG_HOME` is either not set or empty, a default equal to `$HOME/.config` should be used.
> —_XDG Base Directory Specification,_ emphasis mine.

First, my interpretation is that the spec clearly states that:

- `$XDG_CONFIG_HOME` is the only place where user specific files should be stored; this is also in-line with what, IMO, is the spirit of that spec: stop the madness of having config files everywhere in the user's home directory.
- `$HOME/.config/` should only be used as a default for `$XDG_CONFIG_HOME`, if the latter is not set or empty.

Second, `$XDG_CONFIG_HOME` can also be used to override the location of the user configuration files for a specific program, and this seems to like a perfectly valid (if not very common) use case.

If a user chooses to run `XDG_CONFIG_HOME=/some_other_path liquidctl`, they could very well mean that they only want that location to be searched (for the user specific files).  For example, to test how liquidctl behaves with a different configuration than what they normally use, or with no configuration at all.  Sure, there are other ways of doing those things, but my point here is simply that it's _valid_ for a user to achieve them using `$XDG_CONFIG_HOME`.

Reply:
That makes sense and I don't know why, but I interpreted it differently initially, however I now agree with you. 

For some reason I interpreted as `XDG_CONFIG_HOME` should be checked first and if no file is there check `$HOME/.config/` which now that I think about it makes no sense. 

Ill update this later today (and make code code a little better)

Reply:
The counter argument is that a user that just recently set `$XDG_CONFIG_HOME` may still expect `liquidctl` to use his old configuration file, especially if it's in `~/.liquidctl.toml`.

Perhaps we should stop treating `~/.liquidctl.toml` and `~/.config/liquidctl/config.toml` like they are the same thing: the former could always be an override, while the latter could only be used as a default for missing of empty `$XDG_CONFIG_HOME`.